core/src/main/java/kafka/server/share/SharePartitionManager.java

```diff
@@ -532,7 +542,7 @@ List<TopicIdPartition> cachedTopicIdPartitionsInShareSession(String groupId, Uui
 
     // Add the share fetch request to the delayed share fetch purgatory to process the fetch request if it can be
     // completed else watch until it can be completed/timeout.
-    private void addDelayedShareFetch(DelayedShareFetch delayedShareFetch, Set<DelayedShareFetchKey> keys) {
+    private void addDelayedShareFetch(DelayedShareFetch delayedShareFetch, Set<Object> keys) {
         delayedShareFetchPurgatory.tryCompleteElseWatch(delayedShareFetch,
             CollectionConverters.asScala(keys).toSeq().indices());
```

Comment:
I don't think we require `.indices` now, can you please check once.

Reply:
I have followed comment https://github.com/apache/kafka/pull/17380#discussion_r1790671430 and now keys are back to the datatype `Set<DelayedShareFetchKey>`, hence I guess this comment is redundant now.